### PR TITLE
feat: configurable SSA per Dependent Resource

### PR DESCRIPTION
**Pull request by csviri**

### CODE REVIEW

File: operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/kubernetes/KubernetesDependent.java [hunk 1]
```diff
@@ -76,4 +76,9 @@
    * Creates the resource only if did not exist before, this applies only if SSA is used.
    */
   boolean createResourceOnlyIfNotExistingWithSSA() default KubernetesDependentResourceConfig.DEFAULT_CREATE_RESOURCE_ONLY_IF_NOT_EXISTING_WITH_SSA;
+
+  /**
+   * In case of UNDEFINED inherits global configuration.
+   */
+  BooleanWithUndefined useSSA() default BooleanWithUndefined.UNDEFINED;
```
Comment (metacosm) [inline operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/kubernetes/KubernetesDependent.java:83]: Why not use a `Boolean` instead? If the value is `null` then inherit?

File: sample-operators/webpage/src/main/java/io/javaoperatorsdk/operator/sample/dependentresource/ConfigMapDependentResource.java [hunk 2]
```diff
@@ -35,6 +35,7 @@ protected ConfigMap desired(WebPage webPage, Context<WebPage> context) {
     data.put("index.html", webPage.getSpec().getHtml());
     Map<String, String> labels = new HashMap<>();
     labels.put(SELECTOR, "true");
+    log.info("!!! Desired spec: {}", webPage.getSpec().getHtml());
```
Comment (metacosm) [inline sample-operators/webpage/src/main/java/io/javaoperatorsdk/operator/sample/dependentresource/ConfigMapDependentResource.java:38]: That seems irrelevant to this PR

Comment (csviri) [reply sample-operators/webpage/src/main/java/io/javaoperatorsdk/operator/sample/dependentresource/ConfigMapDependentResource.java:38]: yes will remove

File: operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/kubernetes/KubernetesDependent.java (hunk 1 above)
Comment (metacosm) [reply operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/kubernetes/KubernetesDependent.java:83]: Ah yes, d'oh! 🤦🏼

**Review by metacosm — APPROVED** [review]
